### ISSUE TITLE
Cannot push multiple commits to same existing branch

### ISSUE THREAD

**Issue by anechol** [tokens-studio/figma-plugin#1687]
Currently, it seems to be impossible to push multiple commits to the same existing branch with an open pull request. When attempting to do this, there is no error or warning stating this to be the case. There is also no confirmation that the commit was successfully pushed.

For additional context, the network request can be seen from the dev tools, and we can see the new sha for the commit, but checking the branch in GH does not show that commit going through.

Reproduction steps:

1. Add new token or remove existing token
2. Click "Push to GitHub"
3. Enter commit message and add name for a new branch
4. Push and open draft pull request
5. Add another new token and click "Push to Github"
6. Enter commit message and choose the branch name from step 2
7. Push

The new commit doesn't seem to get added to the existing branch here, but as mentioned above, checking the network tab in the dev tools shows the commit going through.

Is it expected behavior for there to always be a fresh branch created to open a pull request?

**Comment by six7** [comment]
Not sure i follow, can you post reproduction steps or a video?

**Comment by anechol** [comment]
Update on this issue here. It seems like the inability to push multiple commits to the same branch comes down to permissions. We've added a bot account's token to allow it to make new commits and pushes. The initial commit and push worked fine, but subsequent commits and pushes to the same branch were "lost to the void." We've discovered that any account making these commits and pushes needs to have admin access to the repo, which the bot account did not have initially, only write access. With those changed permissions, subsequent commits were pushed up to branches as expected.
